service/src/test/java/bio/terra/workspace/service/job/JobServiceLogTest.java

```diff
@@ -0,0 +1,192 @@
+          .email("stairway@unit.com")
+          .token(Optional.of("not-a-real-token"));
+
+  @Autowired private JobService jobService;
+  @Autowired private WorkspaceDao workspaceDao;
+  @Autowired private WorkspaceActivityLogDao activityLogDao;
+  @MockBean private SamService mockSamService;
+
+  @BeforeEach
+  @SuppressFBWarnings(value = "DE_MIGHT_IGNORE", justification = "Mockito flakiness")
+  void setup() {
+    try {
+      Mockito.doReturn(true).when(mockSamService.isAuthorized(any(), any(), any(), any()));
+    } catch (Exception e) {
+      // How does a mock even throw an exception?
+    }
+  }
+
+  /**
+   * Reset the {@link JobService} {@link FlightDebugInfo} after each test so that future submissions
+   * aren't affected.
+   */
+  @AfterEach
+  void clearFlightDebugInfo() {
+    jobService.setFlightDebugInfoForTest(null);
+  }
+
+  @Test
+  void setChangedActivityWhenFlightCreateComplete() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var nullChangedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNull(nullChangedDate);
+
+    runFlight("a creation flight", workspaceUuid, OperationType.CREATE);
+    var changedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNotNull(changedDate);
+  }
+
+  @Test
+  void setChangedActivityWhenFlightUpdateComplete() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var nullChangedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNull(nullChangedDate);
+
+    runFlight("a creation flight", workspaceUuid, OperationType.UPDATE);
+    var changedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNotNull(changedDate);
+  }
+
+  @Test
+  void setChangedActivityWhenFlightDeleteComplete() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var nullChangedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNull(nullChangedDate);
+
+    runFlight("a creation flight", workspaceUuid, OperationType.DELETE);
+    var changedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNotNull(changedDate);
+  }
+
+  @Test
+  void setChangedActivityWhenFlightCloneComplete_notSet() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var nullChangedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNull(nullChangedDate);
+
+    runFlight("a creation flight", workspaceUuid, OperationType.CLONE);
+    var changedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNotNull(changedDate);
+  }
+
+  @Test
+  void setChangedActivityWhenFlightUnknownComplete() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var nullChangedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNull(nullChangedDate);
+
+    runFlight("a creation flight", workspaceUuid, OperationType.UNKNOWN);
+    var changedDate = activityLogDao.getLastChangedDate(workspaceUuid);
+    assertNull(changedDate);
+  }
+
+  @Test
+  void setChangedActivityWhenFlightCreateFails_notSet() {
```

Comment:
Add test for deletion flight fails cases?

Reply:
more tests are added for logging that doesn't happen in the flight.